packages/api/__tests__/fetchOptions.test.ts

```diff
@@ -0,0 +1,115 @@
+import { Api } from '../src';
+import { IApiTestsListData } from './types';
+
+const { TEST_WORKSPACE_ID } = process.env;
+
+describe('fetch options', () => {
+  let originalFetch;
+  let api: Api;
+
+  beforeAll(() => {
+    originalFetch = fetch;
+    // @ts-ignore
+    fetch = jest.fn((...args) => originalFetch(...args));
+  });
+
+  beforeEach(() => {
+    api = new Api({
+      workspaceId: TEST_WORKSPACE_ID,
+    });
+  });
+
+  afterEach(() => {
+    // @ts-ignore
+    fetch.mockClear();
+
+    api = null;
+  });
+
+  afterAll(() => {
+    // @ts-ignore
+    fetch = originalFetch;
+  });
+
+  it('passes options from request method', () => {
+    api.request<IApiTestsListData>(
+      `
+      query {
+        apiTestsList {
+          items {
+            id
+            timestamp
+          }
+        }
+      }
+    `,
+      {},
+      { cache: 'no-cache' },
+    );
+
+    expect(fetch).lastCalledWith(
+      expect.any(String),
+      expect.objectContaining({
+        body: expect.any(String),
+        method: expect.any(String),
+        headers: expect.any(Object),
+        cache: 'no-cache',
+      }),
+    );
+  });
+
+  it("doesn't overwrite body option", () => {
+    const customBody = JSON.stringify({ query: 'Wrong Query', variables: {} });
+    api.request<IApiTestsListData>(
+      `
+      query {
+        apiTestsList {
+          items {
+            id
+            timestamp
+          }
+        }
+      }
+    `,
+      {},
+      // @ts-ignore
+      { body: customBody },
+    );
+
+    expect(fetch).lastCalledWith(
+      expect.any(String),
+      expect.objectContaining({
+        body: expect.not.stringContaining(customBody),
+        method: expect.any(String),
+        headers: expect.any(Object),
+      }),
+    );
+  });
+
+  it("doesn't overwrite method option", () => {
+    api.request<IApiTestsListData>(
+      `
+      query {
+        apiTestsList {
+          items {
+            id
+            timestamp
+          }
+        }
+      }
+    `,
+      {},
```

Comment:
@vorobeez Does the request method template passed variables yet? Should it validate if a variable in the template is messing or a variable passed is un used?

Reply:
I can implement that by query's ast tree parsing and validation.

Reply:
Nevermind! Saw it below.